runtime/service.go

```diff
@@ -627,6 +627,13 @@ func (s *service) StopVM(requestCtx context.Context, request *proto.StopVMReques
 		timeout = time.Duration(request.TimeoutSeconds) * time.Second
 	}
 
+	if info, err := s.machine.DescribeInstanceInfo(requestCtx); err != nil {
+		return nil, err
+	} else if *info.State == models.VMStatePaused {
+		s.logger.Debugf("Instance is paused, can't stop VM")
+		return nil, errors.New("VMPaused")
```

Comment:
- You don't have to use Debugf if the string doesn't have any format directives such as `%s`.
- Can you check other error messages we have? I think we are returning a full sentence rather than something like `VMPaused`

Reply:
Added in most recent commit